examples/nirfsg/rfsg-multitone-example.py

```diff
@@ -0,0 +1,335 @@
+The maximum separation is mandated by the instrument used.
+
+The gRPC API is built from the C API. NI-RFSG documentation is installed with the driver at:
+  C:\Program Files (x86)\IVI Foundation\IVI\Drivers\niRFSG\documentation\English\RFSG.chm
+
+Getting Started:
+
+To run this example, install "NI-RFSG Driver" on the server machine:
+  https://www.ni.com/en-us/support/downloads/drivers/download.ni-rfsg.html
+
+For instructions on how to use protoc to generate gRPC client interfaces, see our "Creating a gRPC
+Client" wiki page:
+  https://github.com/ni/grpc-device/wiki/Creating-a-gRPC-Client
+
+Refer to the NI-RFSG gRPC Wiki for the latest C Function Reference:
+  https://github.com/ni/grpc-device/wiki/NI-RFSG-C-Function-Reference
+
+Running from command line:
+
+Server machine's IP address, port number, and physical channel name can be passed as separate
+command line arguments.
+  > python getting-started-single-tone-generation.py <server_address> <port_number> <resource_name>
+If they are not passed in as command line arguments, then by default the server address will be
+"localhost:31763", with "SimulatedRFSG" as the physical channel name.
+"""  # noqa: W505
+
+import math
+import sys
+
+import grpc
+import nidevice_pb2 as nidevice_grpc
+import nirfsg_pb2 as nirfsg_types
+import nirfsg_pb2_grpc as grpc_nirfsg
+import numpy as np
+
+
+SERVER_ADDRESS = "localhost"
+SERVER_PORT = "31763"
+SESSION_NAME = "NI-RFSG-Session"
+
+# Resource name, channel name and options for a simulated 5841 client.
+RESOURCE = "5840_1"
+OPTIONS = ""
+
+# Read in cmd args
+if len(sys.argv) >= 2:
+    SERVER_ADDRESS = sys.argv[1]
+if len(sys.argv) >= 3:
+    SERVER_PORT = sys.argv[2]
+if len(sys.argv) >= 4:
+    RESOURCE = sys.argv[3]
+    OPTIONS = ""
+# Create a gRPC channel + client.
+channel = grpc.insecure_channel(f"{SERVER_ADDRESS}:{SERVER_PORT}")
+client = grpc_nirfsg.NiRFSGStub(channel)
+vi = None
+
+
+def check_for_warning(response, vi):
+    """Print to console if the status indicates a warning."""
+    if response.status > 0:
+        warning_message = client.ErrorMessage(
+            nirfsg_types.ErrorMessageRequest(vi=vi, error_code=response.status)
+        )
+        sys.stderr.write(f"{warning_message.error_message}\nWarning status: {response.status}\n")
+
+
+class Tone:
+    """Class definition for tones.
+
+    Define the tone with relative offset from the carrier frequency in Hz (relative to the
+     generator center frequency).
+    Relative power level in dBc (relative to the generator defined power level).
+    """
+
+    def __init__(self, offset_hz, gain_db):
+        """Construct each tone using a relative offset in Hz and relative power in dBc."""
+        self.offset_hz = offset_hz
+        self.gain_db = gain_db
+
+    def __repr__(self):
+        """Make a string with Tones information."""
+        return "Offset: {0} Hz and Gain: {1} dB".format(self.offset_hz, self.gain_db)
+
+
+def gcd(my_list):

```

Comment:
Depending on the version of numpy, I _think_ you can use numpy's gcd method since you're already including it. According to https://numpy.org/doc/stable/reference/generated/numpy.gcd.html I think the callers would be something like `np.gcd.reduce(my_list)` instead of calling this method.